docs/changelog/next_release/179.breaking.rst

```diff
@@ -0,0 +1,50 @@
+| Change the way of passing ``hwm`` to ``DBReader`` and ``FileDownloader`` classes:
+| **DBReader** before:
+
+.. code-block:: python
+
+    reader = DBReader(
+        connection=...,
+        source=...,
+        hwm_column=("col1", "cast(col1 as date)"),
+    )
+
+after:
+
+.. code-block:: python
+
+    reader = DBReader(
+        connection=...,
+        source=...,
+        hwm=DBReader.AutoHWM(
+            entity="col1",
+            expression="cast(col1 as date)",
+            name="...",
+        ),
+    )
+
+| **FileDownloader** before:
+
+.. code-block:: python
+
+    downloader = FileDownloader(
+        connection=...,
+        source_path=...,
+        target_path=...,
+        hwm_type="file_list",
+    )
+
+after:
+
+.. code-block:: python
+
+    downloader = FileDownloader(
+        connection=...,
+        source_path=...,
+        target_path=...,
+        hwm=FileListHWM(name="..."),
+    )
+
+| Now user can pass custom hwm classes into ``hwm`` field in ``DBReader`` or ``FileDownloader``
+| Classes from etl_entities.old_hwm are marked as **deprecated**
+| The use of the ``hwm_column`` parameter in ``DBReader``, as ``hwm_type`` in ``FileDownloader`` are deprecated since version 0.10.0. They will be removed in future versions
```

Comment:
```suggestion
| The use of the ``hwm_column`` / ``hwm_expression`` parameters in ``DBReader``, as ``hwm_type`` in ``FileDownloader`` are deprecated since version 0.10.0. They will be removed in future versions
```